### PR TITLE
BI-8848 Implement option filter on cert options page

### PR DESCRIPTION
**Pull request by dlgroves-ch**
* Filter good standing option if company is in liquidation.
* Filter liquidators option if company is active.
* Tested locally; working.

Resolves:
[BI-8848](https://companieshouse.atlassian.net/browse/BI-8848)

### CODE REVIEW

File: test/controller/certificates/llp-certificates/options.controller.unit.test.ts [hunk 1]
```diff
@@ -32,7 +32,14 @@ describe("llp.certificate.options.controller.unit", () => {
             const returnedItemOptions = setItemOptions(options);
 
             chai.expect(returnedItemOptions?.memberDetails?.includeBasicInformation).to.be.true;
-        });            
+        });
+
+        it("should set includeBasicInformation on MemberDetails to true, when the option is liquidators", () => {
```
Comment (phudson-ch) [inline test/controller/certificates/llp-certificates/options.controller.unit.test.ts:37]: Should read: should set includeBasicInformation on LiquidatorsDetails...

File: src/controllers/certificates/options.controller.ts [hunk 2]
```diff
@@ -126,6 +134,12 @@ export const setItemOptions = (options: string[]): ItemOptionsRequest => {
                 itemOptionsAccum.includeCompanyObjectsInformation = true;
                 break;
             }
+            case LIQUIDATORS_FIELD: {
+                //if(FEATURE_FLAGS.liquidatedCompanyCertificatesEnabled()){
+                    itemOptionsAccum.liquidatorsDetails = { includeBasicInformation: true };
```
Comment (phudson-ch) [inline src/controllers/certificates/options.controller.ts:139]: Missing property: add liquidatorsDetails property to ItemOptionsRequest

**Review by phudson-ch — CHANGES_REQUESTED** [review]

File: src/controllers/certificates/options.controller.ts (hunk 2 above)
Comment (dlgroves-ch) [reply src/controllers/certificates/options.controller.ts:139]: TODO: If company status is liquidation, add field to ItemOptionsRequest.

**Review by phudson-ch — APPROVED** [review]

File: src/controllers/certificates/llp-certificates/options.controller.ts [hunk 3]
```diff
@@ -20,14 +23,20 @@ export const render = async (req: Request, res: Response, next: NextFunction): P
...
         const SERVICE_URL = replaceCompanyNumber(LLP_ROOT_CERTIFICATE, certificateItem.companyNumber)
         logger.info(`Certificate item retrieved, id=${certificateItem.id}, user_id=${userId}, company_number=${certificateItem.companyNumber}`);
         return res.render(LLP_CERTIFICATE_OPTIONS, {
             companyNumber: certificateItem.companyNumber,
             itemOptions: certificateItem.itemOptions,
             templateName: LLP_CERTIFICATE_OPTIONS,
-            SERVICE_URL
+            SERVICE_URL,
+            filterMappings: {
+                goodStanding: companyProfile.companyStatus != 'liquidation',
```
Comment (dlgroves-ch) [inline src/controllers/certificates/llp-certificates/options.controller.ts:36]: Use enum constants here.

File: src/controllers/certificates/llp-certificates/options.controller.ts [hunk 4]
```diff
@@ -86,6 +97,9 @@ export const setItemOptions = (options: string[]): ItemOptionsRequest => {
             includeAddressRecordsType: null
         }
     };
+    if(companyStatus === "liquidation") {
```
Comment (dlgroves-ch) [inline src/controllers/certificates/llp-certificates/options.controller.ts:100]: Use enum constant here.

File: src/controllers/certificates/llp-certificates/options.controller.ts [hunk 5]
```diff
@@ -105,6 +119,12 @@ export const setItemOptions = (options: string[]): ItemOptionsRequest => {
                 itemOptionsAccum.memberDetails = { includeBasicInformation: true };
                 break;
             }
+            case LIQUIDATORS_FIELD: {
+                if(companyStatus === "liquidation"){
```
Comment (dlgroves-ch) [inline src/controllers/certificates/llp-certificates/options.controller.ts:123]: Use enum constant here.

File: src/controllers/certificates/options.controller.ts [hunk 6]
```diff
@@ -20,14 +23,20 @@ export const render = async (req: Request, res: Response, next: NextFunction): P
...
         const SERVICE_URL = `/company/${certificateItem.companyNumber}/orderable/certificates`;
         logger.info(`Certificate item retrieved, id=${certificateItem.id}, user_id=${userId}, company_number=${certificateItem.companyNumber}`);
         return res.render(CERTIFICATE_OPTIONS, {
             companyNumber: certificateItem.companyNumber,
             itemOptions: certificateItem.itemOptions,
             templateName: CERTIFICATE_OPTIONS,
-            SERVICE_URL
+            SERVICE_URL,
+            filterMappings: {
+                goodStanding: companyProfile.companyStatus != 'liquidation',
```
Comment (dlgroves-ch) [inline src/controllers/certificates/options.controller.ts:36]: Use enum constant here.

File: src/controllers/certificates/options.controller.ts [hunk 7]
```diff
@@ -101,8 +112,11 @@ export const setItemOptions = (options: string[]): ItemOptionsRequest => {
             includeBasicInformation: null,
             includeAddress: null,
             includeAppointmentDate: null
-        }
+        },
     };
+    if(companyStatus === "liquidation") {
```
Comment (dlgroves-ch) [inline src/controllers/certificates/options.controller.ts:117]: Use enum constant here.

File: src/controllers/certificates/options.controller.ts [hunk 8]
```diff
@@ -126,6 +140,12 @@ export const setItemOptions = (options: string[]): ItemOptionsRequest => {
                 itemOptionsAccum.includeCompanyObjectsInformation = true;
                 break;
             }
+            case LIQUIDATORS_FIELD: {
+                if(companyStatus === "liquidation"){
```
Comment (dlgroves-ch) [inline src/controllers/certificates/options.controller.ts:144]: Use enum constant here.

File: test/controller/certificates/options.controller.unit.test.ts [hunk 9]
```diff
@@ -7,70 +7,103 @@ describe("certificate.options.controller.unit", () => {
...
+
+            chai.expect(returnedItemOptions?.secretaryDetails?.includeBasicInformation).to.be.null;
+            chai.expect(returnedItemOptions?.includeGoodStandingInformation).to.be.null;
+            chai.expect(returnedItemOptions?.includeCompanyObjectsInformation).to.be.null;
+            chai.expect(returnedItemOptions?.directorDetails?.includeBasicInformation).to.be.null;
+            chai.expect(returnedItemOptions?.registeredOfficeAddressDetails?.includeAddressRecordsType).to.be.null;
+            chai.expect(returnedItemOptions).to.not.have.property("liquidatorsDetails");
+        });
+
+        it("should leave multiple itemOptions to null if company is active and options are undefined", () => {
```
Comment (dlgroves-ch) [inline test/controller/certificates/options.controller.unit.test.ts:87]: Fix wording

File: test/controller/certificates/llp-certificates/options.controller.unit.test.ts [hunk 10]
```diff
@@ -7,60 +7,91 @@ describe("llp.certificate.options.controller.unit", () => {
...
             chai.expect(returnedItemOptions?.includeGoodStandingInformation).to.be.null;
             chai.expect(returnedItemOptions?.registeredOfficeAddressDetails?.includeAddressRecordsType).to.be.null;
             chai.expect(returnedItemOptions?.designatedMemberDetails?.includeBasicInformation).to.be.null;
             chai.expect(returnedItemOptions?.memberDetails?.includeBasicInformation).to.be.null;
+            chai.expect(returnedItemOptions).to.not.have.property("liquidatorsDetails");
         });
 
         it("should leave multiple itemOptions to null, when there options are undefined", () => {
-            const returnedItemOptions = setItemOptions(undefined as unknown as string[]);
+            const returnedItemOptions = setItemOptions("active");
```
Comment (dlgroves-ch) [inline test/controller/certificates/llp-certificates/options.controller.unit.test.ts:78]: Fix wording of test case.

**Review by phudson-ch — APPROVED** [review]

**Review by kinpang-CH — APPROVED** [review]

**Pull request merged**